### PR TITLE
Fix 1198 peers file

### LINKED ISSUE #1198: Auto-reconnect seed node when all seed nodes crash
**Summary**
with NEO2.x network, all consensus nodes communicate with each other with seed nodes, in the worst case, all seed nodes crash, this would happen in a very low possibility, then all seed nodes loss connection with consensus nodes, when they are brought back online again, they can not reconnect to the consensus nodes, because consensus nodes can't build a new connection with the seed nodes.
To rebuild the connection, it will requires all consensus nodes to restart.

Reproduction steps:
1. prepare 4 consensus nodes.
2. prepare 2 seed nodes.
3. start seed nodes
4. start 4 consensus nodes.
5. start two normal neo-cli clients.
6. shut down all seed nodes.
7. start seed nodes again.

屏幕快照 2019-10-29 下午7 04 12

**Do you have any solution you want to propose?**
We should add a policy to auto reconnect to the seed nodes when they are all down and back again, and then there is no need to restart all consensus nodes.

**Where in the software does this update applies to?**
- Network Policy
- P2P (TCP)

### PR DESCRIPTION
**Pull request by ShawnYun**
close #1198

### CODE REVIEW

**Comment by erikzhang** [comment]
Most of the nodes saved in peers.dat are offline when you next launch. This way you will spend a lot of time trying to connect to nodes that don't exist. This method has been used before, and later found that the effect is very poor, so it was removed.

**Comment by ShawnYun** [comment]
@erikzhang It will only save at most 10 ConnectedPeers in peers.dat. So it won't spend much time. And it will help seed nodes reconnect to other nodes when they are restarted.

**Comment by erikzhang** [comment]
For seed nodes, the most important thing is to connect to each other, otherwise it is prone to islands. So the first thing to do after the seed node starts is to connect to other seed nodes.

**Comment by erikzhang** [comment]
On the other hand, it is rare for all seed nodes to crash. There is no need to add a global mechanism for this situation.

**Comment by vncoelho** [comment]
In fact, I think this is not so bad if it is optional and not default.

File: neo/Network/P2P/Peer.cs [hunk 1]
```diff
@@ -172,6 +188,64 @@ private void OnStart(ChannelsConfig config)
...
+                {
+                    string[] s = reader.ReadString().Split(':');
+                    IPEndPoint peer;
+                    try
+                    {
+                        peer = GetIPEndpointFromHostPort(s[0], int.Parse(s[1]));
+                    }
+                    catch (AggregateException)
+                    {
+                        continue;
```
Comment (lock9) [inline neo/Network/P2P/Peer.cs]: Shouldn't we log when this happens? How can this happen?

**Comment by ShawnYun** [comment]
@lock9 Hello. For a normal node, it could ask seed nodes to get more peers. But for a seed node, if all the other seed nodes crashed, it couldn't find any peer to connect. The problem is more important for seed nodes, sanving the recent connectedpeers to a file is a soluction, maybe we could find a better soluction.

**Comment by erikzhang** [comment]
We don't need to consider for all seed nodes crashed. Deploying the seed nodes on the cloud and running the monitor is the right way to solve this problem.

**Pull request closed without merging**